### PR TITLE
Customer request

### PR DESCRIPTION
**Pull request by hmasila**
# What does this PR do?

Handles most customer methods
- [ ] creates customer class
- [ ] get_state
- [ ] update_tags
- [ ] delete_tags
- [ ] get_tags
- [ ] add_reminder
- [ ] cancel_reminder
- [ ] get_secondary_ids
- [ ] update_secondary_ids
- [ ] delete_secondary_ids
- [ ] get_metadata
- [ ] update_metadata
- [ ] delete_metadata
- [ ] update_app_data
- [ ] delete_app_data
- [ ] lease_app_data
- [ ] update_activity
- [ ] send_message
- [ ] adopt_state
- [ ] update_messaging_consent
- [ ] reply_to_message

### CODE REVIEW

File: lib/elarian.rb [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
-require_relative "elarian/client"
+require "elarian/requester"
+require "elarian/client"
+require "elarian/customer"
+require "elarian/response_parser"
+require "elarian/utils/utils"
 
 module Elarian
   class Error < StandardError; end
+  EventMachine.run do
```
Comment (aksalj) [inline lib/elarian.rb:17]: What does this section do? Why is it included?

File: lib/elarian/client.rb [hunk 2]
```diff
@@ -1,42 +1,94 @@
...
   class Client
-    def initialize(org_id:, app_id:, api_key:, options: {})
+    EXPECTED_EVENTS = %w[pending error connecting connected closed].freeze
+
+    def initialize(org_id:, app_id:, api_key:, is_sandbox: false, simplex_mode: false, options: {})
       @org_id = org_id
       @app_id = app_id
       @api_key = api_key
+      @simplex_mode = simplex_mode
+      @is_sandbox = is_sandbox
```
Comment (aksalj) [inline lib/elarian/client.rb:12]: `is_simulator` should be the only param used to indicate a simulator connection, not `is_sandbox`, no?

**Review by aksalj — CHANGES_REQUESTED** [review]
Can you include test cases that cover all the methods in the SDK spec?

File: lib/elarian.rb (hunk 1 above)
Comment (hmasila) [reply lib/elarian.rb]: We were using it for testing. I have removed it

**Pull request merged**